src/utils/KML.js

```diff
@@ -157,6 +224,26 @@ const writeFeatures = (layer, featureProjection) => {
       zIndex: styles[0].getZIndex(),
     };
 
+    // Set temporary ID.
+    clone.setId(`id${clone.ol_uid}`);
+
+    // Create array of rotation values for LabelTexts.
+    if (newStyle.text) {
```

Comment:
It seems the extendedData must written in the KML automatically according to the OL code :https://github.com/openlayers/openlayers/blob/v5.3.0/src/ol/format/KML.js#L2687